test/client/views/settings.js

```diff
@@ -10,51 +11,127 @@ var internals = {};
+        this.view = new SettingsView({ settingsModel: this.fakeModel });
+        this.view.render();
+    });
+
     describe('when the client id changes', function () {
 
-        it('updates the jquery snippet');
+        it('updates the jquery snippet', function () {
+
+            this.view.$el.find('.client').val('foo123').keyup();
+
+            expect(this.view.$('.jquery-snippet').text()).to.contain('var clientId = \'foo123\';');
+        });
 
     });
 
     describe('when submit is clicked', function () {
 
-        it('sets the client id on the model');
+        describe('after the client id was changed', function () {
+
+            beforeEach(function () {
+
+                this.view.$el.find('.client').val('foo123').keyup();
+
+                this.view.$('.submit').click();
+            });
 
-        context('with a changed client id', function () {
+            it('sets the client id on the model', function () {
 
-            it('sets the client id as the channel on the model');
+                expect(this.view.settingsModel.get('clientId')).to.equal('foo123');
+            });
 
+            it('sets the client id as the channel on the model', function (){
+                expect(this.view.settingsModel.get('channel')).to.equal('foo123');
+            });
         });
 
-        it('hides itself');
+
+        it('hides itself', function(){
+
+            this.view.$el.find('.submit').click();
+
+            var viewHideSpy = sinon.spy(this.view, 'hide');
+
+            this.view.$el.find('.client').val('foo123').keyup();
+
+            this.view.$el.find('.submit').click();
+
+            expect(viewHideSpy).to.have.been.called;
+        });
 
     });
 
     describe('when cancel is clicked', function () {
 
-        it('resets the views client id back to original client id');
+        it('resets the views client id back to original client id', function () {
+
+            this.view.$el.find('.client').val('foo123').keyup();
+
+            this.view.$el.find('.cancel').click();
+
+            expect(this.view.model.get('clientId')).to.equal('bar123');
+        });
+
+        it('hides itself', function () {
+
+            var viewHideSpy = sinon.spy(this.view, 'hide');
+
+            this.view.$el.find('.cancel').click();
+
+            expect(viewHideSpy).to.have.been.called;
+        });
+
+        it('re-renders itself to reset it\'s display for the next time it\'s shown', function () {
 
-        it('hides itself');
+            var viewRenderSpy = sinon.spy(this.view, 'render');
 
-        it('re-renders itself to reset it\'s display for the next time it\'s shown');
+            this.view.$el.find('.cancel').click();
+
+            expect(viewRenderSpy).to.have.been.called;
+        });
 
     });
 
     describe('when the original settings model is changed', function () {
 
-        it('changes the view\'s model');
+        it('changes the view\'s model', function () {
```

Comment:
This could be split into two tests:

when the original settings model channel changes
  it updates the view's model's channel

when the original settings model client id changes
  it updates the view's model's client id